app-store-connect/src/main.rs

```diff
@@ -0,0 +1,469 @@
+use anyhow::Result;
+use app_store_connect::bundle_api::{BundleId, BundleIdPlatform};
+use app_store_connect::certs_api::{self, Certificate, CertificateType};
+use app_store_connect::device_api::Device;
+use app_store_connect::profile_api::{Profile, ProfileType};
+use app_store_connect::{AppStoreConnectClient, UnifiedApiKey};
+use clap::{Parser, Subcommand};
+use std::io::Write;
+use std::path::PathBuf;
+
+#[derive(Parser)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    #[clap(subcommand)]
+    command: Commands,
+}
+
+fn main() -> Result<()> {
+    env_logger::init();
+    let args = Args::parse();
+    args.command.run()
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Generates a PEM encoded RSA2048 signing key
+    GenerateKey {
+        /// Path to unified api key.
+        #[clap(long)]
+        api_key: PathBuf,
+        /// Certificate type can be one of development, distribution or notarization.
+        #[clap(long)]
+        r#type: String,
+        /// Path to write a new PEM encoded RSA2048 signing key
+        pem: PathBuf,
```

Comment:
apple-codesign supports multiple signing key sources. e.g. YubiKey. It would be nice if we could support other storage mechanisms. But this can be deferred to a follow-up. But it would be nice to unify the terminology if possible in order to avoid a behavior break to unify in the future.

Reply:
well it's consistent with xbuild. in xbuild we use two kinds of objects. `--pem` and `--api-key`, where pem includes a private key and certificate and api key is a unified json object.

Reply:
Ah, I see. This is fine then. We can always add more complexity later.